api/v1beta1/disruption_webhook.go

```diff
@@ -211,3 +225,148 @@ func (r *Disruption) getMetricsTags() []string {
+}
+
+// safetyNetCountNotTooLarge is the safety net regarding the count of targets
+// it will check against the number of targets being targeted and the number of targets in the k8s system
+// > 66% of the k8s system being targeted warrants a safety check if we assume each of our targets are replicated
+// at least twice. > 80% in a namespace also warrants a safety check as namespaces may be shared between services.
+// returning true indicates the safety net caught something
+func safetyNetCountNotTooLarge(r Disruption) (bool, string, error) {
+	if r.Spec.Unsafemode != nil && r.Spec.Unsafemode.DisableCountTooLarge {
+		return false, "", nil
+	}
+
+	userCount := r.Spec.Count
+	totalCount := 0
+	namespaceCount := 0
+	targetCount := 0
+	namespaceThreshold := 0.8
+	clusterThreshold := 0.66
+
+	if r.Spec.Unsafemode.Config != nil && r.Spec.Unsafemode.Config.CountTooLarge != nil {
+		if r.Spec.Unsafemode.Config.CountTooLarge.NamespaceThreshold != 0 {
+			namespaceThreshold = float64(r.Spec.Unsafemode.Config.CountTooLarge.NamespaceThreshold) / 100.0
+		}
+
+		if r.Spec.Unsafemode.Config.CountTooLarge.ClusterThreshold != 0 {
+			clusterThreshold = float64(r.Spec.Unsafemode.Config.CountTooLarge.ClusterThreshold) / 100.0
+		}
+	}
+
+	if r.Spec.Level == chaostypes.DisruptionLevelPod {
+		pods := &corev1.PodList{}
+		listOptions := &client.ListOptions{
+			Namespace: r.ObjectMeta.Namespace,
+		}
+
+		err := k8sClient.List(context.Background(), pods, listOptions)
+		if err != nil {
+			return false, "", fmt.Errorf("error listing namespace pods: %w", err)
+		}
+
+		namespaceCount = len(pods.Items)
+
+		listOptions = &client.ListOptions{
+			Namespace:     r.ObjectMeta.Namespace,
+			LabelSelector: labels.SelectorFromValidatedSet(r.Spec.Selector),
+		}
+
+		err = k8sClient.List(context.Background(), pods, listOptions)
+		if err != nil {
+			return false, "", fmt.Errorf("error listing target pods: %w", err)
+		}
+
+		targetCount = len(pods.Items)
+
+		err = k8sClient.List(context.Background(), pods)
+		if err != nil {
+			return false, "", fmt.Errorf("error listing cluster pods: %w", err)
+		}
+
+		totalCount = len(pods.Items)
+	} else {
+		nodes := &corev1.NodeList{}
+
+		err := k8sClient.List(context.Background(), nodes)
+		if err != nil {
+			return false, "", fmt.Errorf("error listing target pods: %w", err)
+		}
+
+		totalCount = len(nodes.Items)
+	}
+
+	userCountVal := 0.0
+
+	userCountInt, isPercent, err := GetIntOrPercentValueSafely(userCount)
+	if err != nil {
+		return false, "", fmt.Errorf("failed to get count: %w", err)
+	}
+
+	if isPercent {
+		userCountVal = float64(userCountInt) / 100.0 * float64(targetCount)
+	} else {
+		userCountVal = float64(userCountInt)
+	}
+
+	// we check to see if the count represents > 80 percent of all pods in the existing namepsace
+	// or if the count represents > 66 percent of all pods in the cluster
+	if userNamespacePercent := userCountVal / float64(namespaceCount); userNamespacePercent > namespaceThreshold {
```

Comment:
We need to skip this check when `level == node`. Otherwise, we end up dividing by zero, and get a validation error like:
```
	* target selection represents +Inf % of the total pods in the namespace while the threshold is 80.00 %
```